go/tools/builders/nolint.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import "strings"
+
+type Linters []string
+
+func (l Linters) Contains(s string) bool {
```

Comment:
How many different linters does a typical project use? I am wondering whether a `map[string]bool` would be more efficient (and whether that difference actually matters).

Reply:
I use about 40 in my project and did not see a noticeable performance issue.

Reply:
This is not scale against the analyzers inside a project, it scales against the `//nolint:<comma separated list of string>` so we don't expect too bad of a search time here, as a typical user should not be ignoring a billion linters in a directive comment.

With that said, I think it would be cleaner/better if `Linters` is constructed as a Set: `map[string]struct{}` because we will be performing this lookup inside 4-5 levels of nested loops for "each compilation action". It would also save us from needing a type alias as it's very straight forward to do a key lookup in a map.